File: js/math/gcd_lcm.js
// Print the divisor and multiple of fixed pairs, with Euclid's algorithm run on a destructured swap.
// Run: node gcd_lcm.js

const pairs = [
  [1071, 462],
  [270, 192],
  [17, 5],
  [120, 36],
];

function gcd(first, second) {
  while (second !== 0) {
    [first, second] = [second, first % second];
  }
  return first;
}

for (const [first, second] of pairs) {
  const divisor = gcd(first, second);
  console.log(`${first} ${second} ${divisor} ${(first / divisor) * second}`);
}

File: kotlin/math/GcdLcm.kt
// Print the divisor and multiple of fixed pairs, with Euclid's algorithm run on a destructured swap.
// Run: kotlinc GcdLcm.kt -include-runtime -d gcd_lcm.jar && java -jar gcd_lcm.jar

val PAIRS = listOf(1071L to 462L, 270L to 192L, 17L to 5L, 120L to 36L)

fun euclid(first: Long, second: Long): Long {
    var a = first
    var b = second
    while (b != 0L) {
        val remainder = a % b
        a = b
        b = remainder
    }
    return a
}

fun main() {
    for ((first, second) in PAIRS) {
        val divisor = euclid(first, second)
        println("$first $second $divisor ${first / divisor * second}")
    }
}
